text/0003-measure-metric-type.md

```diff
@@ -2,50 +2,169 @@
+| Unit | The unit of measurement being recorded. |
+| Description | Information about this metric. |
+
+See the specification for more information on these fields, including formatting and uniqueness requirements.  To define a new metric, use one of the language-specific API methods (e.g., with names like `NewCumulativeMetric`, `NewGaugeMetric`, or `NewMeasureMetric`).
+
+Metric instrument Handles are SDK-provided objects that combine a metric instrument with a set of pre-defined labels.  Handles are obtained by calling a language-specific API method (e.g., `GetHandle`) on the metric instrument with its label values.  Handles may be used to `Set()`, `Add()`, or `Record()` metrics according to their kind.  The `Set()`, `Add()`, and `Record()` 
+
+## Selecting Metric Kind
+
+By separation of API and implementation in OpenTelemetry, we know that an implementation is free to do _anything_ in response to a metric API call.  By the low-level interpretation defined above, all metric events have the same structural representation, only their logical interpretation varies according to the metric definition.  Therefore, we select metric kinds based on two primary concerns:
+
+1. What should be the default implementation behavior?  Unless configured otherwise, how should the implementation treat this metric variable?
+1. How will the program source code read?  Each metric uses a different verb, which helps convey meaning and describe default behavior.  Cumulatives have an `Add()` method.  Gauges have a `Set()` method.  Measures have a `Record()` method.
+
+To guide the user in selecting the right kind of metric for an application, we'll consider the following questions about the primary intent of reporting given data.  We use "of primary interest" here to mean information that is almost certainly useful in understanding system behavior.  Consider these questions:
+
+- Does the measurement represent a quantity of something?  Is it also non-negative?
+- Is the sum a matter of primary interest?
+- Is the event count a matter of primary interest?
+- Is the distribution (p50, p99, etc.) a matter of primary interest?
+
+The specification will be updated with the following guidance.
+
+### Cumulative metric
+
+Likely to be the most common kind of metric, cumulative metric events express the computation of a sum.  Choose this kind of metric when the value is a quantity, the sum is of primary interest, and the event count and distribution are not of primary interest.  To raise (or lower) a cumulative metric, call the `Add()` method.
+
+If the quantity in question is always non-negative, it implies that the sum is strictly ascending.  When this is the case, the cumulative metric also serves to define a rate.  For this reason, cumulative metrics have a `NonNegative` option to be declared as non-negative.  The API will reject negative updates to non-negative cumulative metrics, instead submitting an SDK error event, which helps ensure meaningful rate calculations.
```

Comment:
Excuse me for being a nitpicker, but if quantity is zero then it is still non-negative, but the sum will be ascending, not strictly ascending. So either the quantity is positive for the sum to be strictly ascending or the quantity is non-negative for the sum to be ascending.

Reply:
@krnowak a rate can be calculated for any ascending sum (not necessary strictly ascending). @jmacd please fix :)

Reply:
Yes, thanks. Also, I thought some more and believe that this non-negative case should be the default case. This left me wondering about a different name that reflects the opposite property, that the cumulative can go up and down. The phrase "Up-Down Counter" exists, but I also like "Bidirectional" to describe this. How do you feel about these changes?